HW2/P4/driver.py

```diff
@@ -27,6 +28,65 @@ def py_median_3x3(image, iterations=10, num_threads=1):
 
     return tmpA
 
+# my code with parallelize thread
+
+def process_lines(tmpA, tmpB, no_thread,num_threads,locks):
+    # process  no. line 
+    line = no_thread
+    while  line  < tmpA.shape[0] :       
```

Comment:
I get that you are using locks to handle writing conflicts by acquiring and releasing the locks that correspond to the threads above and below the current line. However, this may be fairly expensive consider the locking and unlocking happen frequently. A good alternative is to use event handlers.